cdap-storage-spi/src/main/java/io/cdap/cdap/spi/data/StructuredTable.java

```diff
@@ -273,6 +273,17 @@ void increment(Collection<Field<?>> keys, String column, long amount)
    */
   void deleteAll(Range keyRange) throws InvalidFieldException, IOException;
 
+  /**
+   * Updates the specific fields in a range of rows from the table
+   *
+   * @param keyRange key range of the rows to update: cab only be a primary key prefix
+   * @param fields the fields to write to
+   * @throws InvalidFieldException if any of the keys are not part of table schema,
+   *                               or their types do not match the schema
+   * @throws IOException if there is an error reading from or updating the table
+   */
+  void updateAll(Range keyRange, Collection<Field<?>> fields) throws InvalidFieldException, IOException;
```

Comment:
nit, wondering if `updateRange` sounds more proper?

Reply:
yup agree

Reply:
however we have deleteAll that also deletes range - so in terms of consistency it is better?

Reply:
yah it makes sense to be consistent